### PR TITLE
fix: Move logout and local date preference logic to organizer vm

### LINKED ISSUE #1128: Move logout and local date preference out of Main Presenter
### Chore

**Description**

Move logout and local date preference out of Main Presenter and put it in OrganizerViewModel. Delete the MainPresenter. Also add tests for the View models.

**Would you like to work on the issue?**

Yes

**Parent issue**

#894

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #1128

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: Move logout and local date preference out of Main Presenter and put it in OrganizerViewModel. Delete the MainPresenter. Also add tests for the View models.

Screenshots for the change: Just to show that everything works as before.
videotogif_2018 06 26_03 05 50

### CODE REVIEW

**Comment by srv-twry** [comment]
@Masquerade0097 @sridharjajoo @mishuvs please review.

**Comment by sridharjajoo** [comment]
@srv-twry Its working fine. I have tested it

File: app/src/main/java/org/fossasia/openevent/app/core/main/OrganizerViewModel.java [hunk 1]
```diff
@@ -4,37 +4,82 @@
...
     private final CompositeDisposable compositeDisposable = new CompositeDisposable();
 
     private final MutableLiveData<User> organizer = new MutableLiveData<>();
+    private final MutableLiveData<String> error = new MutableLiveData<>();
+    private final SingleEventLiveData<Void> logoutAction = new SingleEventLiveData<>();
+    private final SingleEventLiveData<Void> localDatePreferenceAction = new SingleEventLiveData<>();
 
     @Inject
-    public OrganizerViewModel(UserRepository userRepository) {
+    public OrganizerViewModel(UserRepository userRepository, AuthService loginModel,
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/main/OrganizerViewModel.java]: authService

**Comment by srv-twry** [comment]
@iamareebjamal Please review.

**Pull request merged**